pkg/codegen/docs/static_schema_loader.go

```diff
@@ -0,0 +1,49 @@
+package docs
+
+import (
+	"fmt"
+
+	"github.com/blang/semver"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+type staticSchemaLoader struct {
+	schema           *schema.Package
+	schemaReferences map[string]*schema.Package
+}
+
+var _ schema.ReferenceLoader = (*staticSchemaLoader)(nil)
+
+func NewStaticSchemaLoader(loadedSchema *schema.Package) schema.ReferenceLoader {
+	// TODO: resolve schema references from input schema
```

Comment:
Is this for things like awsx? Do we need a tracking issue for this?

Reply:
Yeah but even as is, it still works for some languages iirc. There is follow-up work to get more resources and packages to generate constructor examples. I will open issues and link them back here once this is merged ✅